moves.pl

```diff
@@ -70,10 +74,34 @@
     other_player(PlayerColor, OtherPlayer),
     slot_legend(OtherPlayerBall, OtherPlayer).
 
-% TODO
-% % alters board state and player scores if necessary
-% make_move(move(FirstBall, Direction)).
 
-% % TODO
-% % a "move" object, defines a legal move
-% move(BoardState, PlayerColor, RowIndex, ColIndex, Direction, NextBoardState).
+killer_move(PlayerColor, [PlayerBall, PlayerBall, PlayerBall, OtherPlayerBall, OtherPlayerBall, Border | _]):-
+    validate_colors(PlayerColor, PlayerBall,OtherPlayerBall, _, Border).
+
+killer_move(PlayerColor, [PlayerBall, PlayerBall, PlayerBall, OtherPlayerBall, Border | _]):-
+    validate_colors(PlayerColor, PlayerBall,OtherPlayerBall, _, Border).
+
+killer_move(PlayerColor, [PlayerBall, PlayerBall, OtherPlayerBall, Border | _]):-
+    validate_colors(PlayerColor, PlayerBall,OtherPlayerBall, _, Border).
+
+% Move (can be used to make move by passing NextBoardState result back to game handler)
+move(BoardState, PlayerColor, RowIndex, ColIndex, Direction, NextBoardState):-
+    board_size(BoardSize),
+    legal_move(PlayerColor, BoardState, RowIndex, ColIndex, Direction),!,
+    slots_sequence_by_direction(BoardState, BoardSize, RowIndex, ColIndex, Direction, EffectedSlotsState),
+    move_slots_forward_in_line(PlayerColor, EffectedSlotsState, NextEffectedSlotsState),
+    generate_changed_board(BoardState, RowIndex, ColIndex, Direction, NextEffectedSlotsState, NextBoardState).
+
+move_slots_forward_in_line(PlayerColor, CurrSlotsState, NextSlotsState):-
+    member(0, CurrSlotsState), 
+    (
+        append([MovedBalls, [0], StaticBalls], CurrSlotsState), !,
```

Comment:
I think some comments are needed for each case here

Reply:
👍